pubsub/gcp/pubsub/pubsub.go

```diff
@@ -174,12 +214,20 @@ func (g *GCPPubSub) Publish(ctx context.Context, req *pubsub.PublishRequest) err
 	if g.closed.Load() {
 		return errors.New("component is closed")
 	}
+	g.lock.RLock()
+	_, topicExists := g.topicCache[req.Topic]
+	g.lock.RUnlock()
 
-	if !g.metadata.DisableEntityManagement {
+	if !g.metadata.DisableEntityManagement && !topicExists {
```

Comment:
Just adding a comment here for future people:

```suggestion
	// We are not acquiring a write lock before calling ensureTopic, so there's the chance that ensureTopic be called multiple times
    // This is acceptable in our case, even is slightly wasteful, as ensureTopic is idempotent
	if !g.metadata.DisableEntityManagement && !topicExists {
```